File: 剑指offer/二叉树/二叉树的镜像/code.py
# -*- coding:utf-8 -*-


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.right = None
        self.left = None


class Solution:
    def mirror(self, root):
        if not root:
            return None
        tmp = self.mirror(root.right)
        root.right = self.mirror(root.left)
        root.left = tmp
        return root

File: 剑指offer/二叉树/二叉树的镜像/code.cpp
struct TreeNode {
	int val;
	struct TreeNode* left;
	struct TreeNode* rught;
	TreeNode(int x):
		val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
	int TreeDepth(TreeNode* pRoot)
	{
		if (!pRoot) return 0;
		return max(TreeDepth(TreeNode->left)+1, TreeDepth(TreeNode->right)+1)
	}
};